pkg/blockbuilder/config.go

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package blockbuilder
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/grafana/mimir/pkg/storage/tsdb"
+)
+
+type Config struct {
+	InstanceID          string             `yaml:"instance_id" doc:"default=<hostname>" category:"advanced"`
+	PartitionAssignment map[string][]int32 `yaml:"partition_assignment" category:"experimental"`
+	DataDir             string             `yaml:"data_dir"`
+
+	ConsumeInterval       time.Duration `yaml:"consume_interval"`
+	ConsumeIntervalBuffer time.Duration `yaml:"consume_interval_buffer"`
+	LookbackOnNoCommit    time.Duration `yaml:"lookback_on_no_commit" category:"advanced"`
+
+	Kafka KafkaConfig `yaml:"kafka"`
+
+	// BlocksStorageConfig is defined outside the block-builder config and is injected dynamically.
+	BlocksStorageConfig tsdb.BlocksStorageConfig `yaml:"-"`
+}
```

Comment:
Didn't we agree to use the same configuration from ingest storage - `ingest_strage.kafka`? Why are there yaml tags on the `kafka` field?

Reply:
I'm doing it in the next PR #9147